plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/property/build/WebHooksPropertySource.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.property.build;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.ui.views.properties.IPropertyDescriptor;
+import org.eclipse.ui.views.properties.IPropertySource;
+import org.eclipse.ui.views.properties.PropertyDescriptor;
+
+import com.openshift3.client.model.build.IBuildTrigger;
+import com.openshift3.client.model.build.IWebhookTrigger;
+
+/**
+ * Property source to display a BuildConfig's webhook triggers.
+ */
+public class WebHooksPropertySource implements IPropertySource{
+
+	private IWebhookTrigger genericTrigger;
+	private IWebhookTrigger gitTrigger;
+
+	public WebHooksPropertySource(Collection<IBuildTrigger> triggers ){
+		for (IBuildTrigger trigger : triggers) {
+			switch(trigger.getType()){
+			case generic:
+				genericTrigger = (IWebhookTrigger) trigger;
+				break;
```

Comment:
isnt this just processing the first element out of serverals and leaving the rest alone?